addons/stock_account/models/stock_location.py

```diff
@@ -30,3 +32,13 @@ def _should_be_valued(self):
         if self.usage == 'internal' or (self.usage == 'transit' and self.company_id):
             return True
         return False
+
+    @api.depends('usage')
+    def _compute_valuation_accounts(self):
+        for location in self:
+            if location.usage == 'production' and location.company_id:
+                location.valuation_in_account_id = location.company_id.property_stock_production_account_id
+                location.valuation_out_account_id = location.company_id.property_stock_production_account_id
+            else:
+                location.valuation_in_account_id = False
+                location.valuation_out_account_id = False
```

Comment:
@clesgow we just do it because if the user selects production by mistake then the account is auto set but after he selects any other type then still account is there. and in some cases, this field is invisible so users don't know about it.
this is store compute so compute only called when any change in depends.

and this field is already there so no change in existing data.

Reply:
but still if you think it's not needed to set False then we remove it.

Reply:
While I get what you say, the same could be said with `Inventory Loss` usage that gets set to `False` as well, and those two fields are still visible for this.
What you could do is override the `write()` so you have the previous value of `usage` and know if you need to empty the fields or not.

Reply:
okay